pkg/webhook/cspc.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net/http"
+)
+
+func (wh *webhook) validateCSPC(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+	req := ar.Request
+	response := &v1beta1.AdmissionResponse{}
+	// validates only if requested operation is CREATE or DELETE
+	if req.Operation == v1beta1.Create {
+		glog.V(5).Infof("Admission webhook create request for type %s", req.Kind.Kind)
+		return wh.validateCSPCCreateRequest(req)
+	} else if req.Operation == v1beta1.Update {
```

Comment:
think of moving the frequent passing check to be first.. in this case, it should be 'Update'?
just a food for thought

Reply:
yes.